sdks/java/extensions/python/src/main/java/org/apache/beam/sdk/extensions/python/PythonExternalTransform.java

```diff
@@ -162,6 +167,26 @@ public PythonExternalTransform<InputT, OutputT> withKwargs(Row kwargs) {
     return this;
   }
 
+  /**
+   * Specifies the field type of arguments.
+   *
+   * <p>Type hints are especially useful for logical types since type inference does not work well
+   * for logical types.
+   *
+   * @param argType A class object for the argument type.
+   * @param fieldType A schema field type for the argument.
+   * @return updated wrapper for the cross-language transform.
+   */
+  public PythonExternalTransform<InputT, OutputT> withTypeHint(
```

Comment:
Should this be per arg instead of per type ? In other words, can the same class map to different schema types ?

Reply:
Per type makes more sense to me. Do you have any specific per arg use-case in mind?

Reply:
I don't. Just wasn't sure. We can keep this per type if Robert and Brian are OK.